mgc/dscr/oneSample.py

```diff
@@ -0,0 +1,110 @@
+from sklearn.metrics import euclidean_distances
+from ._utils import _CheckInputs
+import numpy as np
+import random
+from .base import DiscriminabilityTest
+from scipy._lib._util import MapWrapper
+
+
+
+class oneSample(DiscriminabilityTest):
+    r"""
+     A class that performs a one-sample test for whether the discriminability differs from random chance,
+     as described in [1]. With :math:`D_X` as the sample discriminability of :math:`X`, it tests whether:
+
+     .. math::
+        H_0: D_X = D_0
+
+     and
+
+     .. math::
+        H_A: D_X > D_0 
+    
+    where :math:`D_0` is the discriminability that would be observed by random chance.
+    
+    References
+    ----------
+    .. [#1Dscr] Eric W. Bridgeford, et al. "Optimal Decisions for Reference 
+                Pipelines and Datasets: Applications in Connectomics." Bioarxiv (2019).
+    """
+
+    def __init__(self):
+        self.stat = None
+        self.pvalue = None
+        DiscriminabilityTest.__init__(self)
+
+
+
+    def test(self, X, Y, isDist = False, remove_isolates=True, reps=1000, workers=-1):
```

Comment:
this may be more of a question for @sampan501 but is this where you want parameters of the algorithm to go, or were you going with something more like sklearn where the parameters get passed into the class `__init__()` and then `test` only eats data?

I ask because as written, nothing about this needs to be a class as far as I can tell, if everything is just passed to `test()`.

Reply:
I think that some of these functions, including `isDist` should be `compute_distance` similar to other functions in `mgc`. Maybe also `remove_isolates`. I think that rest of the parameters are fine.

I think that a class should be used to conform to the common API for other modules in `mgc`